tests/lib/eslint.js

```diff
@@ -2488,6 +2488,23 @@ describe("eslint", function() {
             assert.equal(messages[2].column, 18);
         });
 
+        it("should accept a pregenerated ast", function() {
+
+            var code = "foo()\n    alert('test')";
+            var config = { rules: { "no-mixed-spaces-and-tabs": 1, "eol-last": 1, "semi": [1, "always"] } };
+
+            var ast = eslint.parse(code, config);
+
+            var messages = eslint.verify({text: code, ast: ast}, config, filename);
```

Comment:
Note that I am calling eslint.parse (eslint.js:481), rather than calling esprima.parse directly. That function calls esprima with all the output options on, so the "ast" does include a tokens array.